### PR TITLE
feat(release-automation): add update-winget-manifest command

### PR DESCRIPTION
**Pull request by hatayama**
## Summary

- Add a release automation command that opens winget-pkgs submissions for stable dispatcher releases.
- Generate the version, installer, and locale manifests from existing release metadata and assets.

## User Impact

- Stable Windows releases can be submitted to winget through a repeatable automation path.
- Missing credentials, prereleases, existing upstream versions, existing branches, and open pull requests are handled safely for release retries.

## Changes

- Render the three manifests required for `hatayama.uloop` using schema version `1.10.0`.
- Sync the configured fork, create or reuse a release branch, update manifest files, and open the upstream pull request through GitHub CLI APIs.
- Add focused tests for rendering, skip paths, initial and update titles, branch retries, and duplicate pull requests.

## Verification

- `go test ./internal/automation/ -run Winget -v`
- `scripts/check-go-cli.sh`
- `env -u WINGET_PKGS_TOKEN go run ./cmd/update-winget-manifest --repo hatayama/unity-cli-loop --tag dispatcher-v3.1.0 --fork-repo hatayama/winget-pkgs`

<a href="https://cubic.dev/pr/hatayama/unity-cli-loop/pull/2484?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: cli/release-automation/internal/automation/winget_manifest_update.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	if token == "" {
+		writeWingetManifestUpdateLine(stdout, "WINGET_PKGS_TOKEN is not configured; skipping winget manifest update.")
+		return 0
+	}
+
+	version, err := dispatcherVersionFromReleaseTag(config.tag)
+	if err != nil {
+		return writeWingetManifestUpdateError(stderr, err)
+	}
+	if strings.Contains(version, "-") {
```
[the remark this answers is not in the document]
Comment (hatayama) [reply cli/release-automation/internal/automation/winget_manifest_update.go:147]: Fixed in `2ef508634`. The existing release metadata request now fetches `publishedAt,isPrerelease`, while the tag-based hyphen check remains first. A stable-looking release marked as a GitHub prerelease exits successfully with a distinct skip message and performs no manifest PUT or pull-request creation.

[the remark this answers is not in the document]
Comment (hatayama) [reply cli/release-automation/internal/automation/winget_manifest_update.go:147]: Fixed in `2ef508634`. The existing release metadata request now fetches `publishedAt,isPrerelease`, while the tag-based hyphen check remains first. A stable-looking release marked as a GitHub prerelease exits successfully with a distinct skip message and performs no manifest PUT or pull-request creation.

File: cli/release-automation/internal/automation/winget_manifest_github.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	forkRepo string,
+	branch string,
+	path string,
+) (string, error) {
+	output, err := deps.runOutput(
+		ctx,
+		wingetTokenEnvironment(token),
+		"gh",
+		"api",
+		"repos/"+forkRepo+"/contents/"+path+"?ref="+branch,
```
[the remark this answers is not in the document]
Comment (hatayama) [reply cli/release-automation/internal/automation/winget_manifest_github.go]: Fixed in `2ef508634`. Branch values used by the fork-content `ref` query and upstream pull-request `head` query are now encoded with `url.QueryEscape`. The regression test covers a `3.1.0+build` version and verifies `%2B` encoding.

File: cli/release-automation/internal/automation/winget_manifest_update.go [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+		writeWingetManifestUpdateLine(stdout, fmt.Sprintf("dispatcher %s is a pre-release; winget receives stable releases only. Skipping.", version))
+		return 0
+	}
+
+	data, err := loadWingetManifestData(ctx, deps, config, version)
```
[the remark this answers is not in the document]
Comment (hatayama) [reply cli/release-automation/internal/automation/winget_manifest_update.go]: Fixed in `2ef508634`. The upstream version existence check now runs before release metadata and asset retrieval. The existing-version test verifies both release download and release view are called zero times, with no manifest PUT or pull-request creation.
